tests/unit/test_subspaces.py

```diff
@@ -28,7 +28,13 @@ def test_dimension(self):
     def test_mapping_single(self):
         self.assertEqual(Full._idx_to_state(np.array(10, dtype=dnm_int_t), 5), 10)
         self.assertEqual(Full._state_to_idx(np.array(10, dtype=dnm_int_t), 5), 10)
-
+    
+    def test_mapping_invalid_i2s(self):
+        L = 5
+        dim = Full._get_dimension(L)
+        with self.assertRaises(ValueError): 
+            state = Full._idx_to_state(np.array([-1, 2**(L)], dtype=dnm_int_t), L)
```

Comment:
use `with self.subTest(idx=idx)` to check both negative and positive cause errors